pkg/services/extsvcauth/registry/service.go

```diff
@@ -7,38 +7,38 @@ import (
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/infra/slugify"
 	"github.com/grafana/grafana/pkg/services/extsvcauth"
-	"github.com/grafana/grafana/pkg/services/extsvcauth/oauthserver"
+	"github.com/grafana/grafana/pkg/services/extsvcauth/oauthserver/oasimpl"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	"github.com/grafana/grafana/pkg/services/serviceaccounts/extsvcaccounts"
 )
 
 var _ extsvcauth.ExternalServiceRegistry = &Registry{}
 
 type Registry struct {
-	features    featuremgmt.FeatureToggles
-	logger      log.Logger
-	oauthServer oauthserver.OAuth2Server
-	saSvc       *extsvcaccounts.ExtSvcAccountsService
+	features featuremgmt.FeatureToggles
+	logger   log.Logger
+	oauthReg extsvcauth.ExternalServiceRegistry
+	saReg    extsvcauth.ExternalServiceRegistry
 
 	extSvcProviders map[string]extsvcauth.AuthProvider
 	lock            sync.Mutex
 }
 
-func ProvideExtSvcRegistry(oauthServer oauthserver.OAuth2Server, saSvc *extsvcaccounts.ExtSvcAccountsService, features featuremgmt.FeatureToggles) *Registry {
+func ProvideExtSvcRegistry(oauthServer *oasimpl.OAuth2ServiceImpl, saSvc *extsvcaccounts.ExtSvcAccountsService, features featuremgmt.FeatureToggles) *Registry {
 	return &Registry{
 		extSvcProviders: map[string]extsvcauth.AuthProvider{},
 		features:        features,
 		lock:            sync.Mutex{},
 		logger:          log.New("extsvcauth.registry"),
-		oauthServer:     oauthServer,
-		saSvc:           saSvc,
+		oauthReg:        oauthServer,
+		saReg:           saSvc,
 	}
 }
 
 // HasExternalService returns whether an external service has been saved with that name.
-func (r *Registry) HasExternalService(ctx context.Context, name string) bool {
+func (r *Registry) HasExternalService(ctx context.Context, name string) (bool, error) {
```

Comment:
For this function I was not sure whether I needed to fallback to `oauthReg` and `saReg`, but thought this saved a round trip to the DB.
Normally we will clean up orphaned services in another PR so the list is in phase with the DB.